openwisp_notifications/base/views.py

```diff
@@ -0,0 +1,40 @@
+from django.contrib.auth import get_user_model
+from django.contrib.auth.mixins import LoginRequiredMixin, UserPassesTestMixin
+from django.http import Http404
+from django.views.generic import TemplateView
+
+User = get_user_model()
+
+
+class NotificationSettingPage(LoginRequiredMixin, UserPassesTestMixin, TemplateView):
+    template_name = 'openwisp_notifications/settings.html'
+    login_url = '/admin/login/'
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        user_id = self.kwargs.get('pk')
+
+        if user_id:
+            try:
+                user = User.objects.get(pk=user_id)
+                # Only admin should access other users settings
+                if not self.request.user.is_staff:
+                    raise Http404("You do not have permission to access this page.")
```

Comment:
maintain the use of single quotes for consistency please

Reply:
Resolved.